assets/shared/data/store-helpers.js

```diff
@@ -25,3 +25,21 @@ export const composeFetchAction = (
 			yield { type: errorAction, error };
 		}
 	};
+
+/**
+ * Create reducer from a map of action type keys and reducer function.
+ *
+ * @example
+ *  createSimpleReducer({ SAMPLE_ACTION: ( { actionProperty }, state ) => ({ ...state, actionProperty }) )
+ *
+ * @param {Object} reducers     Map of action type - reducer functions.
+ * @param {Object} defaultState Default state.
+ *
+ * @return {Function} Store reducer.
+ */
+export const createReducerFromActionMap = ( reducers, defaultState ) => {
+	return ( state = defaultState, action ) => {
+		const reducer = reducers[ action.type ] || reducers.DEFAULT;
+		return reducer( action, state );
```

Comment:
WDYT to invert the `action` and `state` to follow the reducers standard? If yes, we should invert in the `reducers` too.

Reply:
I did this intentionally, as the `action` is the main parameter the reducers will use - sometimes even ignoring state, so they could be simpler, like: `SET_JOB: ( { job } ) => ( { job } )` instead of `SET_JOB: ( state, { job } ) => ( { job } )`.

Reply:
My concert about that was cause confusion while developing because the standard for redux or for the `useReducer` is inverted, but okay for me. 😉